pkg/storage/stores/shipper/compactor/retention/marker.go

```diff
@@ -259,7 +281,7 @@ func (r *markerProcessor) processPath(path string, deleteFunc func(ctx context.C
 	if err != nil {
 		return err
 	}
-	dbUpdate.MaxBatchDelay = 1 * time.Second // 1 s is way enough for saving changes, worst case this operation is idempotent.
+	dbUpdate.MaxBatchDelay = 5 * time.Millisecond
```

Comment:
Should this be configurable maybe?

Reply:
I would not prefer making this configurable. It would just add confusion with more flags, and I have seen people do crazy things with configs when they are not able to operate loki. We can change it if we feel it is not able to keep up in a large cluster.

Reply:
Yeah I actually made that mistake of increasing it. I vote for not making it changeable. In most case it would suffice, otherwise you can already increase the amount of worker to fight against this bottleneck.

Here increasing this will force all goroutine to wait at least the delay for saving their deletion of a key.

Reply:
Thanks for the explanation fellas :+1: